File: cpp/02_library/source/src/main/cpp/namespace/StringGenerator.cpp
${{VAR_COPYRIGHT_HEADER}}

#include <string>

#include "${{VAR_NAMESPACE_PATH}}/StringGenerator.h"

${{VAR_NAMESPACE_DECL_BEGIN}}
using std::string;

StringGenerator::StringGenerator()
    :count(0) { }


string StringGenerator::generate() {
    return "${{VAR_PROJECT_SLOGAN_STRING}} " + std::to_string(count++);
}

${{VAR_NAMESPACE_DECL_END}}

File: java/02_library/source/src/main/java/namespace/StringGenerator.java
${{VAR_COPYRIGHT_HEADER}}

${{VAR_NAMESPACE_PACKAGE_DECLARATION}}

/**
 * A dummy implementation of the {@link Generator} interface.
 *
 */
public class StringGenerator implements Generator<String> {

    private String val;
    private int n;

    /**
     * Constructs a new <code>StringGenerator</code> for generating dummy strings
     */
    public StringGenerator(){
        this.val = "${{VAR_PROJECT_SLOGAN_STRING}}";
    }

    /**
     * Generates a dummy String of this StringGenerator
     *
     * @return A String object of this Generator
     */
    @Override
    public String generate(){
        return this.val + " " + (n++);
    }

}
